pixeltable/globals.py

```diff
@@ -600,18 +601,34 @@ def create_dir(
         Create a directory and replace if it already exists:
 
         >>> pxt.create_dir('my_dir', if_exists='replace_force')
+
+        Create a sub directory along with its ancestors:
+
+        >>> pxt.create_dir('parent1.parent2.sub_dir', create_parents=True)
     """
     path_obj = catalog.Path(path)
     cat = Catalog.get()
 
     with env.Env.get().begin_xact():
+        parent = cat.get_schema_object(str(path_obj.parent), raise_if_parent_missing=(not create_parents))
+        if parent is None and create_parents:
+            # start walking from the root
+            current_dir_path = catalog.Path('', empty_is_valid=True)
+            current_dir_object = cat.get_schema_object(str(current_dir_path))
+            for component in path_obj.components[:-1]:
```

Comment:
Instead of bolting Paths together manually, evolve the Path class:
- property num_ancestors
- ancestor(n: int) -> Path, where ancestor(0) return parent, ancestor(1) returns the parent's parent, etc.

Reply:
I added ancestors() that generates ancestors starting from root level.